crates/clvm-traits/src/clvm_encoder.rs

```diff
@@ -12,6 +12,23 @@ pub trait ClvmEncoder {
         rest: Self::Node,
     ) -> Result<Self::Node, ToClvmError>;
 
+    fn encode_curried_arg(
+        &mut self,
+        first: Self::Node,
+        rest: Self::Node,
+    ) -> Result<Self::Node, ToClvmError> {
+        let q = self.encode_atom(&[1])?;
+        let c = self.encode_atom(&[4])?;
+
+        let quoted = self.encode_pair(q, first)?;
+        let nil = self.encode_atom(&[])?;
+
+        let op_rest = self.encode_pair(rest, nil)?;
+        let op_args = self.encode_pair(quoted, op_rest)?;
```

Comment:
is it not possible to rely on the `ToClvm` implementation for tuple and primitives here, by saying something like:

```
((1, first), ((4, rest), 0)).encode(self)
```

Reply:
Good point